File: client/java/src/avt/FlyTextInfo.java
package avt;

import javax.microedition.lcdui.Graphics;
import javax.microedition.lcdui.Image;
import main.Canvas;

public final class FlyTextInfo {
   private String text = "";
   private int x;
   private int y;
   private int state;
   private int delay;
   private boolean isSmall = false;
   private Image img;
   private byte dir;
   private byte normal = -1;
   private short imgID = -1;
   private short imgID_2 = -1;

   public FlyTextInfo(int var1, int var2, int var3, int var4, Image var5, int var6, int var7, int var8) {
      this.delay = var6;
      this.dir = (byte)var4;
      this.x = var1;
      this.y = var2;
      if (var3 > 0) {
         this.text = "+";
      }

      this.text = this.text + var3;
      if (var3 == 0) {
         this.text = "";
      }

      this.img = var5;
      this.isSmall = false;
      this.normal = -1;
      this.imgID = (short)var7;
      this.imgID_2 = (short)var8;
   }

   public FlyTextInfo(int var1, int var2, String var3, int var4, int var5, int var6) {
      this.delay = var6;
      this.dir = (byte)var4;
      this.x = var1;
      this.y = var2;
      this.text = var3;
      this.state = 0;
      this.isSmall = true;
      this.normal = (byte)var5;
      this.imgID = -1;
      this.imgID_2 = -1;
   }

   public final void update() {
      if (this.delay > 0) {
         --this.delay;
      } else {
         ++this.state;
         if (this.state > 40) {
            this.img = null;
            Canvas.flyTexts.removeElement(this);
         }

         if (this.state < 3) {
            this.y += -2 * this.dir;
         } else {
            this.y += this.dir;
         }
      }

   }

   public final void paint(Graphics var1) {
      if (Canvas.currentMyScreen == RaceScr.me) {
         Canvas.resetTrans(var1);
      }

      if (this.delay <= 0) {
         int var2 = AvMain.hd;
         if (Canvas.currentMyScreen == BoardScr.me && (BoardScr.isStartGame || BoardScr.disableReady) || Canvas.currentMyScreen == RaceScr.me) {
            var2 = 1;
         }

         FontX var3 = Canvas.O;
         if (this.isSmall) {
            if (this.normal == 0) {
               var3 = Canvas.smallFontRed;
            } else if (this.normal == 2) {
               var3 = Canvas.smallFontYellow;
            } else if (this.normal == 3) {
               var3 = Canvas.R;
            } else {
               var3 = Canvas.borderFont;
            }
         }

         var3.drawString(var1, this.text, this.x * var2, this.y * var2, 2);
         if (this.img == null) {
            if (this.imgID != -1) {
               FarmData.paintImg(var1, this.imgID, this.x * var2, (this.y - 5) * var2, 33);
               return;
            }

            if (this.imgID_2 != -1) {
               AvatarData.paintImg(var1, this.imgID_2, this.x * var2, (this.y - 5) * var2, 33);
               return;
            }
         } else if (!this.isSmall) {
            var1.drawImage(this.img, this.x * var2, this.y * var2, 33);
         }
      }

   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/FlyTextInfo.cs
using UnityEngine;

public class FlyTextInfo
{
	private string text = string.Empty;

	private int x;

	private int y;

	private int state;

	private int delay;

	private bool isSmall;

	private Image img;

	private int imgID = -1;

	private int imgID_2;

	private sbyte dir;

	private sbyte normal = -1;

	public FlyTextInfo(int x0, int y0, int text1, int dir1, Image img1, int delay1, int imgID, int imgID_2)
	{
		delay = delay1;
		dir = (sbyte)dir1;
		x = x0;
		y = y0;
		if (text1 > 0)
		{
			text = "+";
		}
		text += text1;
		if (text1 == 0)
		{
			text = string.Empty;
		}
		img = img1;
		isSmall = false;
		normal = -1;
		this.imgID = imgID;
		this.imgID_2 = imgID_2;
	}

	public FlyTextInfo(int x0, int y0, string text1, int dir1, int type, int delay1)
	{
		delay = delay1;
		dir = (sbyte)dir1;
		x = x0;
		y = y0;
		text = text1;
		state = 0;
		isSmall = true;
		normal = (sbyte)type;
		imgID = -1;
		imgID_2 = -1;
	}

	public void update()
	{
		if (delay > 0)
		{
			delay--;
			return;
		}
		state++;
		if (state > 40)
		{
			img = null;
			Canvas.flyTexts.removeElement(this);
		}
		if (state < 3)
		{
			y += -2 * dir;
		}
		else
		{
			y += dir;
		}
	}

	public void paint(MyGraphics g)
	{
		GUIUtility.ScaleAroundPivot(Vector2.one * AvMain.zoom, Vector2.zero);
		if (Canvas.currentMyScreen == RaceScr.me)
		{
			Canvas.resetTrans(g);
		}
		if (delay > 0)
		{
			GUIUtility.ScaleAroundPivot(Vector2.one / AvMain.zoom, Vector2.zero);
			return;
		}
		int num = AvMain.hd;
		if ((Canvas.currentMyScreen == BoardScr.me && (BoardScr.isStartGame || BoardScr.disableReady)) || Canvas.currentMyScreen == RaceScr.me)
		{
			num = 1;
		}
		if (isSmall)
		{
			if (normal == 0)
			{
				Canvas.smallFontRed.drawString(g, text, x * num, y * num, 2);
			}
			else
			{
				Canvas.borderFont.drawString(g, text, x * num, y * num, 2);
			}
		}
		else
		{
			Canvas.numberFont.drawString(g, text, x * num, y * num, 2);
			if (img == null)
			{
				if (imgID != -1)
				{
					FarmData.paintImg(g, imgID, x * num, (y - 5) * num, MyGraphics.HCENTER | MyGraphics.BOTTOM);
				}
				else if (imgID_2 != -1)
				{
					AvatarData.paintImg(g, imgID_2, x * num, (y - 5) * num, MyGraphics.HCENTER | MyGraphics.BOTTOM);
				}
			}
			else if (img != null)
			{
				g.drawImage(img, x * num, y * num, MyGraphics.BOTTOM | MyGraphics.HCENTER);
			}
		}
		GUIUtility.ScaleAroundPivot(Vector2.one / AvMain.zoom, Vector2.zero);
	}
}
